plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-actions-block/block.tsx

```diff
@@ -30,14 +33,28 @@ const Block = ( {
 	placeOrderButtonLabel: string;
 } ): JSX.Element => {
 	const { paymentMethodButtonLabel } = useCheckoutSubmit();
-	const label = applyCheckoutFilter( {
+
+	const store = select( CART_STORE_KEY );
+	const cartTotals = store.getCartTotals();
+	const totalPrice = formatPrice( cartTotals.total_price );
+
+	let label = applyCheckoutFilter( {
 		filterName: 'placeOrderButtonLabel',
 		defaultValue:
 			paymentMethodButtonLabel ||
 			placeOrderButtonLabel ||
 			defaultPlaceOrderButtonLabel,
 	} );
 
+	if ( label.includes( '<price/>' ) ) {
+		if ( cartTotals.total_price === '0' ) {
+			label = label.replace( '<price/>', '' );
+			label = label.replace( /[^a-zA-Z\s]/g, '' );
+		} else {
+			label = label.replace( '<price/>', totalPrice );
+		}
+	}
+
```

Comment:
Would it make sense to have 2 filters here? one for free value and one for paid one? or should developers assume both in a single label and adjust correctly?

Reply:
This is not a blocking comment

Reply:
> Would it make sense to have 2 filters here? one for free value and one for paid one? or should developers assume both in a single label and adjust correctly?

In p1712921238327049-slack-C8X6Q7XQU, I've asked a similar question. I've asked if it would make sense to add a filter that allows to control whether the price should be displayed or not. It was my understanding that we should be careful when introducing additional filters. 

My current implementation utilises the existing `placeOrderButtonLabel` filter. 

- If a merchant/developer is not using the `<price>` placeholder, no price will be displayed.
- If a merchant/developer is using the `<price>` placeholder, but the total cart price is zero, no price and no divider will be displayed.

Did this answer your question or have you had something different in mind?